docs/DeveloperGuide.md

```diff
@@ -1,14 +1,157 @@
 # Developer Guide
 
-## Design & implementation
+## Design
```

Comment:
Do make sure to fill up your Design section, to provide some brief context for your implementation section. Alternatively, you could also choose to merge your Design and Implementation sections together!